### PR TITLE
Structural Changes and Reformatting

### PR DESCRIPTION
**Pull request by DewgongAgarwal**
changes made to : 

send2
grubeditdeadline
coord_upload
coord_view_grub
ssms_student_table
ssms_grub_batchallocation

added helper function : 

batch_allocation

### CODE REVIEW

**Comment by devenbansod** [comment]
Hi @DewgongAgarwal @twitu nice to see activity on this Org. 

I would love to see some automated unit tests or integration tests written for our softwares. These would really help us out in catching breakages due to such business logic changes. 

I could help out as well, if I'm able to find time.

**Comment by twitu** [comment]
Thanks for the appreciation.
That is exactly what i intend to before deployment next semester.
I also want to integrate this project with the heroku github pipeline.

File: ssms/views.py [hunk 1]
```diff
@@ -1480,492 +1428,114 @@ def allocate(grub, grubstu, no, batchlist):
...
+        batch_list.append(batch)
 
+    grubstu = Grub_Student.objects.filter(
+        gm_id=grub.gm_id, status="Signed Up").order_by('bhawan')
+    allocate(grub, grubstu, n, batch_list) #allocating the batches using the batchlist generated
+
+
+@user_passes_test(lambda user: user.is_superuser, login_url='ssms/coord/login')
 def ssms_grub_batchallocation(request, gmid):
-    if request.user.is_superuser:
```
Comment (rtgdk) [inline ssms/views.py:1486]: You removed this line. Anybody can access this page.
Please remember the access privileges while changing/creating a view.

Apart from that, nice work! Appreciate it. 😄

Comment (twitu) [reply ssms/views.py:1486]: fixed access in send2 and ssms_student_table in https://github.com/SSMS-Pilani/SSMS-Grub-Portal/commit/0e748ee8bcdaf5126f8ad214dc382d459bf5d8af

**Pull request merged**